frontend/packages/client/src/components/StepByStep/index.js

```diff
@@ -73,6 +59,71 @@ function StepByStep({
     [refs]
   );
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const goToStep = useCallback(
+    (step, replace = false) => {
+      searchParams.set('step', step + 1);
+      setSearchParams(searchParams, { replace });
+    },
+    [searchParams, setSearchParams]
+  );
+  const tryToGoForward = useCallback(() => {
+    const runPreCheckStepAdvance = () => {
+      if (refs.current) {
+        const runCheckResult = refs.current();
+        if (!runCheckResult) {
+          return false;
+        }
+        refs.current = null;
+      }
+      return true;
+    };
+    const canAdvance = runPreCheckStepAdvance();
+    if (currentStep + 1 < steps.length && canAdvance) {
+      goToStep(currentStep + 1);
+    }
+  }, [currentStep, goToStep, refs, steps.length]);
+  const tryToGoBack = useCallback(() => {
+    if (currentStep - 1 >= 0) {
+      goToStep(currentStep - 1);
+    }
+  }, [currentStep, goToStep]);
+
+  useEffect(() => {
+    const displayStep = searchParams.get('step');
+    const displayStepNum = Number(displayStep);
+    const validStepsVals = Object.values(validSteps);
+    const numValidSteps = validStepsVals.length;
+    const allValidSteps = () =>
+      numValidSteps &&
+      validStepsVals.slice(0, displayStepNum - 1).every((val) => val === true);
```

Comment:
not sure about this condition, it does not fail when displayStepNum is not a number because it's not evaluated due to the condition `numValidSteps` being 0 at first load. Same as in line 106. Could it be done more pure?